pkg/api/handlers/compat/images_build.go

```diff
@@ -169,14 +170,60 @@ func BuildImage(w http.ResponseWriter, r *http.Request) {
 
 	// convert addcaps formats
 	containerFiles := []string{}
-	if _, found := r.URL.Query()["dockerfile"]; found {
-		var m = []string{}
-		if err := json.Unmarshal([]byte(query.Dockerfile), &m); err != nil {
-			// it's not json, assume just a string
-			m = []string{filepath.Join(contextDirectory, query.Dockerfile)}
+	// If query paramemter `dockerfile` is set and is
+	// on priority never look for `Dockerfile` or
+	// `Containerfile` in the context directory.
+	dockerFileSet := false
+	// Query paramemter `remote` always takes priority
+	// over parameter `dockerfile`.
+	remoteSet := false
+	if utils.IsLibpodRequest(r) && query.Remote != "" {
+		// The context directory could be a URL.  Try to handle that.
+		anchorDir, err := ioutil.TempDir("", "libpod_builder")
```

Comment:
You want this tmpdir to use the default tmpdir

```
anchorDir, err := ioutil.TempDir(utils.GetTmpDir(), "libpod_builder")
```

Reply:
yes swtiched it.